src/common/plugins/plugin-readiness-middleware.js

```diff
@@ -0,0 +1,54 @@
+//@flow
+
+import {BaseMiddleware, getLogger} from '@playkit-js/playkit-js';
+import {BasePlugin} from './base-plugin';
+class PluginReadinessMiddleware extends BaseMiddleware {
+  _plugins: Array<BasePlugin>;
+  id: string = 'PluginReadinessMiddleware';
+  static _logger = getLogger('PluginReadinessMiddleware');
+
+  constructor(plugins: Array<BasePlugin>) {
+    super();
+    this._plugins = plugins;
+    PluginReadinessMiddleware._logger.debug('plugins readiness', this._plugins);
+  }
+
+  /**
+   * Load middleware handler.
+   * @param {Function} next - The load handler in the middleware chain.
+   * @returns {void}
+   * @memberof PluginReadinessMiddleware
+   */
+  load(next: Function): void {
+    this.checkNextSettle(0, next);
+  }
+  checkNextSettle(index, next: Function) {
+    if (index < this._plugins.length) {
+      this.checkSettle(index, next);
+    } else {
+      this.callNext(next);
+    }
+  }
+  checkSettle(index: number, next: Function) {
+    const readyPromise = this._plugins[index].ready ? this._plugins[index].ready : Promise.resolve();
```

Comment:
As we discussed, I thought this is a good protection but in general any plugin using the Kaltura Player that implements this will have this base plugin so there shouldn't be any issue with the base plugin always returning `Promise.resolve()`

Reply:
This is for external plugins